### ISSUE TITLE
Rendering the same ajax form multiple times.

### ISSUE THREAD

**Issue by Youniteus** [verbb/formie#260]
**Description**
If I render the same Ajax form multiple times on the same page, only the first one works (with Ajax).

**Steps to reproduce**
1. Configure a form in the plugin admin panel.
2. Render the same form twice using `{{ craft.formie.renderForm('form') }}`
3. Submit the second form.

### Form settings
Please pick from the options below.

- Multi-page form: No
- Submission Method: Ajax
- Client-side Validation: Yes
- Custom Form Templates: No

**Additional info**
- Plugin version:
- Craft version: 3.5.11.1
- Multi-site: No

**Additional context**
A possible fix would be to be able to override the formId, but that doesn't seem working with `{% do form.setSettings({}) %}`.

**Comment by Youniteus** [comment]
I figured out that I could do:
```    
{% set form = craft.formie.forms.handle('learnMore').one() %}
{% do form.setAttributes({
    id: 11111,
}) %}
{{ craft.formie.renderForm(form) }}

{% set form = craft.formie.forms.handle('learnMore').one() %}
{% do form.setAttributes({
    id: 22222,
}) %}
{{ craft.formie.renderForm(form) }}
```

Though now I have the following issue:
- Trying to submit the second form, validates the first form. -> Works incorrectly.
- Trying to submit the first form, does not validate the second form. -> Works correctly.

**Comment by engram-design** [comment]
Should be fixed for the next release. To get the fix early, change your `verbb/formie` requirement in `composer.json` to:

```
"require": {
  "verbb/formie": "dev-craft-3 as 1.3.3",
  "...": "..."
}
```

Then run `composer update`.

**Comment by Youniteus** [comment]
I just tested it, and it's not working correctly.  

After some debugging I found the issue:
- The validation always targets the page of the form, and not the form itself.
- The ids of the pages of both my forms are the same.

So when you submit the second form, the validation target will be the `id="formie-p-1239"` element, so this will validate the first form instead of the second form.

---
**Possible fixes:**
- Correctly target the form instead of the page for validation.
This would only solve the problem for single-page forms.
```php
// src/web/assets/frontend/src/js/formie-form-theme.js

validate(focus = true) {
    if (!this.validationOnSubmit) {
        return true;
    }

    var $fieldset = this.$form;

    // While testing, this.$currentPage seems always to be set.
    // Adding a condition to the next if-statement to only target the page if there are multiple pages.

    if (this.$currentPage) {
        $fieldset = this.$currentPage;
    }

    [...]
}
```
- Make changes so that pages have a unique id.
Not sure if that would cause other problems, but that would also solve the problem for multi-page forms.
```php
// src/templates/_special/form-template/page.html

// adding #{form.formId} to the page id seems to solve my problems.

{% set attributes = {
    id: "formie-p-#{page.id}-#{form.formId}",
    class: [
        'fui-page',
        form.hasMultiplePages and form.settings.submitMethod == 'ajax' and page != form.getCurrentPage() ? 'fui-hidden',
    ],
    data: {
        index: page.sortOrder,
    },
} %}
```
---
Now I realise I could just use a custom form template to solve my problem, but just wanted to let you know. 😉

**Comment by engram-design** [comment]
Appreciate the feedback! Should be updated in the `craft-3` branch.

**Comment by engram-design** [comment]
Fixed in [1.3.4](https://github.com/verbb/formie/releases/tag/1.3.4)
